### PR TITLE
Adds support for headers & body to web requests. 

### PR DESCRIPTION
**Pull request by serussell**
Configs have two new attributes: `headers` and `body`.  `headers` is a map, which are added to the request as headers. Web configs with a `body` attribute are automatically submitted as `POST`s with the body as the payload.

### CODE REVIEW

File: check.go [hunk 1]
```diff
@@ -199,7 +203,27 @@ func (check *Check) web(name *string, sleep *time.Duration) {
...
 // The actual HTTP GET.
 func (check *Check) fetch() error {
-	resp, err := http.Get(check.Web)
+	var resp *http.Response
+	var err error
+	method := "GET"
+	if len(check.Headers) > 0 {
+		client := http.Client{}
+		var body io.Reader
+		if len(check.Body) > 0 {
```
Comment (chillum) [inline check.go]: I bet, this means that `body` won't work without `headers`. If that's intended (I believe, not), what's the reason?

**Comment by chillum** [comment]
ow, one more thing: could you please edit this pull request to be merged into my devel branch (not the master one). thanks

File: check.go (hunk 1 above)
Comment (serussell) [inline check.go]: Correct, this is an error.  It should be checking for the length of `check.Body`.

We can back out the change for Body; I don't use it myself, and included it only to cover some use cases that might come up.

File: check.go [hunk 2]
```diff
@@ -12,18 +14,20 @@ import (
...
-	Alert  string `json:"-"`
-	Tries  int    `json:"-"`
-	Repeat int    `json:"-"`
-	Sleep  int    `json:"-"`
-	Failed bool   `json:"failed" yaml:"-"`
-	Since  string `json:"since,omitempty" yaml:"-"`
+	Name    string            `json:"name,omitempty"`
+	Web     string            `json:"web,omitempty"`
+	Headers map[string]string `json:"headers,omitempty"`
+	Body    string            `json:"body,omitempty"`
```
Comment (serussell) [inline check.go]: How else do you pass `Authorization` or `Content-Type` headers for a `POST`?

Comment (chillum) [inline check.go]: err, let me be clear:
1. we have a [YAML config](https://github.com/chillum/jsonmon/wiki/Configuration) file that is used for settings checks' parameters
2. we have a [JSON status report](https://github.com/chillum/jsonmon/wiki/RESTful-API) for Web UI

so, again, my question is: why do we need `headers` and `body` in JSON at all?

File: check.go (hunk 1 above)
Comment (chillum) [inline check.go]: to me this approach seems reasonable. we can always add it when we will actually need it. :-)

File: check.go (hunk 2 above)
Comment (serussell) [inline check.go]: Ah, I was wondering why there were JSON tags on everything, and YAML tags on almost nothing.  I thought the yaml library might be doing something "clever."

You're right, there would be no purpose to the json tags.  

You can reject the entire pull request; if I get time to poke at it again I'll submit a new one.

**Pull request closed without merging**